backend/src/main/java/ai/verta/modeldb/versioning/blob/diff/ConflictGenerator.java

```diff
@@ -397,23 +400,28 @@ private static AutogenCodeDiff getCodeConflictBlob(
 
   private static AutogenNotebookCodeDiff getNoteBookDiff(
       AutogenNotebookCodeDiff a, AutogenNotebookCodeDiff b, AutogenNotebookCodeBlob c) {
-    return Utils.removeEmpty(
+
+    AutogenNotebookCodeDiff diff =
         new AutogenNotebookCodeDiff()
             .setGitRepo(
                 getGitDiff(
                     a == null ? null : a.getGitRepo(),
                     b == null ? null : b.getGitRepo(),
-                    c == null ? null : c.getGitRepo()))
-            .setPath(
-                new AutogenPathDatasetComponentDiff()
-                    .setA(a == null || a.getPath() == null ? null : a.getPath().getB())
-                    .setB(b == null || b.getPath() == null ? null : b.getPath().getB())
-                    .setC(c == null ? null : c.getPath())
-                    .setStatus(AutogenDiffStatusEnumDiffStatus.fromProto(DiffStatus.CONFLICTED))));
+                    c == null ? null : c.getGitRepo()));
+    if (((a == null || a.getPath() == null) && (b == null || b.getPath() == null))
+        || (a != null && a.equals(b))) return Utils.removeEmpty(diff);
+    diff.setPath(
```

Comment:
there was a bug where PathDataset would appear as a conflict when not expected

Reply:
I'm not sure I follow what this does and why it's an issue only for notebook paths.

Reply:
bug was the code setting a AutogenPathDatasetComponentDiff without validation. so even if a,b,c are null a stray Status.conflict was present in the conflicts. this is not related directly to the bug you reported , but was identified during the testing

Reply:
Wouldn't this also be an issue for the Git part of the blob?

Reply:
the git part is handled in getGitDiff